web/main/models.py

```diff
@@ -2403,8 +2444,22 @@ def _delete_related_links_and_text_blocks(self):
     def get_slug(self):
         return slugify(self.title)
 
-    def viewable_by(self, user):
-        return self.casebook.viewable_by(user)
+    def viewable_by(self, user: User) -> bool:
+        """A node may belong to a viewable casebook, but itself not be viewable because of professor-only content"""
+
+        if not self.casebook:  # An orphaned node isn't viewable, presumably
+            return False
+
+        # Doesn't matter what the permissions are on this node if the casebook itself is not viewable
+        if not self.casebook.viewable_by(user):
+            return False
+
+        if self.is_instructional_material:
+            return user.is_authenticated and (
+                user.verified_professor or user in self.casebook.all_collaborators
+            )
```

Comment:
Above in `nodes_for_user_by_casebook` you did:
```
            if user.is_authenticated and (
                user.verified_professor
                or casebook.contentcollaborator_set.filter(user=user).exists()
            ):
```
similar to this. I wonder if there's a way to abstract out; maybe not worth it...

Reply:
Yeah it was a tossup for me too. The number of methods on these models is so overwhelming I went with inlining it, but if it has to be added a third time it should probably live somewhere else.